src/grblogtools/header_parser.py

```diff
@@ -0,0 +1,63 @@
+import re
+
+from grblogtools.helpers import ParserFlag
+
+
+class HeaderLogParser:
+    # Each might represent the start of the header parser depending on how Gurobi
+    # is run
+    header_log_starts = [
+        re.compile(
+            "Gurobi (?P<Version>\d{1,2}\.[^\s]+) \((?P<Platform>[^\)]+)\) logging started (?P<Time>.*)$"
+        ),
+        re.compile("Logging started (?P<Time>.*)$"),
+        re.compile(
+            "Gurobi Compute Server Worker version (?P<Version>\d{1,2}\.[^\s]+) build (.*) \((?P<Platform>[^\)]+)\)$"
+        ),
+        re.compile("Compute Server job ID: (?P<JobID>.*)$"),
+    ]
+
+    def __init__(self):
+        """Initialize the log parser for the header.
+
+        Note: All header log is captured in one line.
+        """
+        # The integer value representing the current parser status
+        self._flag = ParserFlag.UNKNOWN.value
+        self._log = {}
+
+    def get_flag(self, line: str) -> str:
+        """Return the current status of the header parser.
+
+        Args:
+            line (str): A line in the log file.
+
+        Returns:
+            str: The current parsing status.
+        """
+        if self._flag == 0:
+            for possible_start in HeaderLogParser.header_log_starts:
+                if match := possible_start.match(line):
```

Comment:
```suggestion
                match = possible_start.match(line)
                if match:
```

Reply:
The walrus operator is only available in Python 3.8+. Supporting 3.7 is more important than saving one line of code IMO.

Reply:
True, thanks @mattmilten - we're still prototyping things a little bit, so some 3.8 semantics might sneak in, but we won't merge anything back until it's fully supporting 3.7.

Or, we wait 18 months until 3.7 end of life ... and then keep the walrus ;)